agent/agent.go

```diff
@@ -26,32 +25,32 @@ type Agent struct {
 }
 
 type Config struct {
-	InstanceName    string
-	SSHAddress      string
-	DiscoveryPeriod time.Duration
-	CollectorConfig *collector.Config
+	InstanceName      string
+	DiscoveriesConfig *discovery.DiscoveriesConfig
 }
 
 // NewAgent returns a new instance of Agent with the given configuration
 func NewAgent(config *Config) (*Agent, error) {
-	collectorClient, err := collector.NewCollectorClient(config.CollectorConfig)
+	collectorClient, err := collector.NewCollectorClient(config.DiscoveriesConfig.CollectorConfig)
 	if err != nil {
 		return nil, errors.Wrap(err, "could not create a collector client")
 	}
 
 	ctx, ctxCancel := context.WithCancel(context.Background())
+
+	var discoveries discovery.DiscoveryList
+	discoveries, err = discoveries.AddDiscovery(discovery.NewClusterDiscovery, collectorClient, *config.DiscoveriesConfig)
```

Comment:
We don't do anything here with the errors, right? So the validation is not really used for anything